src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs

```diff
@@ -12,34 +12,73 @@ namespace Microsoft.PowerShell.Commands
     [OutputType(typeof(System.IO.FileInfo))]
     public class NewTemporaryFileCommand : Cmdlet
     {
+        private string _extension = ".tmp";
```

Comment:
Yes, I meant that
```c#
if (Extension parameter absent)
{
    return  Path.GetFileName()
}
else
{
   Here new code to create a temp file with explicit extension.
}
```
We should follow CoreFX if we can. Here we can. `Extension` parameter will be used rarely so it is good two code paths.